closed/OpenJ9.gmk

```diff
@@ -454,14 +454,47 @@ ifneq (,$(OPENJ9_DEVELOPER_DIR))
 endif
 
 ifeq (true,$(OPENJ9_ENABLE_CMAKE))
+CMAKE_ARGS :=
+
+# We grab the C/C++ compilers detected by autoconf or provided by user, not
+# the CC/CXX variables defined by the makefiles, which potentitally include
+# the ccache command which will throw off cmake
+ifneq (,$(OPENJ9_CC))
+  CMAKE_ARGS += "-DCMAKE_C_COMPILER=$(OPENJ9_CC)"
+else
+  CMAKE_ARGS += "-DCMAKE_C_COMPILER=$(ac_cv_prog_CC)"
+endif
+
+ifneq (,$(OPENJ9_CXX))
+  CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(OPENJ9_CXX)"
+else
+  CMAKE_ARGS += "-DCMAKE_CXX_COMPILER=$(ac_cv_prog_CXX)"
+endif
+
+ifneq (,$(CCACHE))
+  # Open jdk makefiles add a  bunch of environemnt variables to the ccache command
+  # cmake will not parse this properly, so we wrap the whole thing in the env command
+  # We also need to add semicolons between arguments or else cmake will treat the whole
+  # thing as one long command name
+
+  # Note: we remove the CCACHE_COMPRESS option that openjdk adds, because it significantly
+  # slows down the build (to the point of erasing any gains from using ccache)
+  CCACHE_NOCOMPRESS := $(filter-out CCACHE_COMPRESS=1,$(CCACHE))
+  ESCAPED_CCACHE :=env$(shell printf ";%s" $(CCACHE_NOCOMPRESS))
+
+  CMAKE_ARGS += "-DCMAKE_CXX_COMPILER_LAUNCHER=$(ESCAPED_CCACHE)"
+  CMAKE_ARGS += "-DCMAKE_C_COMPILER_LAUNCHER=$(ESCAPED_CCACHE)"
+endif
 $(OUTPUTDIR)/vm/cmake.stamp :
 	cd $(OUTPUTDIR)/vm && \
 	$(CMAKE) \
-	  -C $(OPENJ9_TOPDIR)/runtime/cmake/caches/$(OPENJ9_BUILDSPEC).cmake \
+		-C $(OPENJ9_TOPDIR)/runtime/cmake/caches/$(OPENJ9_BUILDSPEC).cmake \
+		$(CMAKE_ARGS) \
 		-DJ9VM_OMR_DIR=$(OPENJ9OMR_TOPDIR) \
 		-DBUILD_ID=$(BUILD_ID) \
 		-DJAVA_SPEC_VERSION=$(VERSION_MAJOR) \
 		-DOPENJ9_BUILD=true \
+		$(EXTRA_CMAKE_ARGS) \
```

Comment:
Where does `EXTRA_CMAKE_ARGS` come from?

Reply:
User provided into make. It is the equivalent of `EXTRA_CONFIGURE_ARGS`